clearer code for radio

dfilatov opened the pull request:
cc @narqo @veged

dfilatov committed:
clearer code for radio

narqo commented:
Cool, but maybe in addition to your changes we could do a little bit more: 

1) in https://github.com/bem/bem-components/blob/radio-imp/common.blocks/radio-group/radio-group.js#L53-L58 – as I remember, `radio` augments `base-control`, so it should be impossible to set `_focused` if one has `_disabled`
2) https://github.com/bem/bem-components/blob/radio-imp/common.blocks/radio-group/radio-group.js#L124 – guess this variable should be called `radios`
3) Why do we use `forEach` here, https://github.com/bem/bem-components/blob/radio-imp/common.blocks/radio-group/radio-group.js#L139, but `while` in other cases

dfilatov commented:
@narqo 
1) There is another reason to check for `disabled`: I need to find first radio that's able to be focused. 
I've remade a little bit that code and I've added test, take a look.
2) Fixed
3) Because I need to break cycle depending on condition, but using of `some()` for that purpose is unacceptable for me.

narqo commented:
Oh, now I see, tnx. Looks clear
